pkg/engine/utils.go

```diff
@@ -256,13 +256,18 @@ func matchSubjects(ruleSubjects []rbacv1.Subject, userInfo authenticationv1.User
 }
 
 //MatchesResourceDescription checks if the resource matches resource description of the rule or not
-func MatchesResourceDescription(resourceRef unstructured.Unstructured, ruleRef kyverno.Rule, admissionInfoRef kyverno.RequestInfo, dynamicConfig []string, namespaceLabels map[string]string) error {
+func MatchesResourceDescription(resourceRef unstructured.Unstructured, ruleRef kyverno.Rule, admissionInfoRef kyverno.RequestInfo, dynamicConfig []string, namespaceLabels map[string]string, policyNamespace string) error {
 
 	rule := *ruleRef.DeepCopy()
 	resource := *resourceRef.DeepCopy()
 	admissionInfo := *admissionInfoRef.DeepCopy()
 
 	var reasonsForFailure []error
+	if policyNamespace != "" {
+		if policyNamespace != resourceRef.GetNamespace() {
+			return errors.New("")
+		}
+	}
```

Comment:
Why two if conditions. it can be done in single if condition. 
```
if policyNamespace != "" && policyNamespace != resourceRef.GetNamespace() {
 	return errors.New("")
}
```

Is there an error or not ? If it's not an error then send `nil`, if it's an error send an proper error message.

Reply:
yes, it can be converted into single `if` condition. Regarding error,  I don't know what error to return.  Because when we apply the namespace policy in Kyverno cluster, suppose in `testing` and if resource is applied in `production` namespace then we don't get any error. Similarly the same case here too, that  resource namespace doesn't match with policy namespace.  
So, either we can return `nil` or any kind of msg like:-  that your resource of `production` namespace doesn't lie in Policy namespace i.e. `testing`.
What's your suggestion @NoSkillGirl ?

Reply:
If you want to tell the user something went wrong...you should return the error.